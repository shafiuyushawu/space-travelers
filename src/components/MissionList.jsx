import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { missionStatus } from '../redux/missions/missionSlice';

const MissionList = ({ mission }) => {
  const dispatch = useDispatch();

  const changeStatus = () => {
    dispatch(missionStatus({ missionId: mission.mission_id }));
  };

  return (
    <tr>
      <th className="p-2 border">{mission.mission_name}</th>
      <td className="p-2 border ">
        <p>{mission.description}</p>
      </td>
      <td className="p-2 border w-36">
        <span
          className={
            mission.member
              ? 'bg-[#00e5ff] font-bold rounded-xl p-1 text-white'
              : 'bg-slate-500 font-bold rounded-xl p-1 text-white'
          }
        >
          {mission.member ? 'Active Member' : 'Not A Member'}
        </span>
      </td>
      <td className="p-2 border w-36">
        <button
          onClick={changeStatus}
          type="button"
          className={
            mission.member
              ? '  border-2 border-[#ff0044] text-[#ff0044] font-bold p-1'
              : 'border-2 border-[#adaaab] text-[#adaaab] font-bold p-1'
          }
        >
          {mission.member ? 'Leave Mission' : 'Join Mission'}
        </button>
      </td>
    </tr>
  );
};

MissionList.propTypes = {
  mission: PropTypes.shape({
    mission_name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    mission_id: PropTypes.string.isRequired,
    member: PropTypes.bool.isRequired,
  }).isRequired,
};

export default MissionList;
