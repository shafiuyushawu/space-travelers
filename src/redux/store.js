import { configureStore } from '@reduxjs/toolkit';
import missionSlice from './missions/missionSlice';
import rocketReducer from './rockets/rocketsSlice';

const store = configureStore({
  missions: missionSlice,
  rockets: rocketReducer,
});
export default store;
