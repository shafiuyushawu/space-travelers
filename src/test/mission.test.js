import { render, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MissionList from '../components/MissionList';

afterEach(cleanup);

describe('Testing mission components', () => {
    test('Generating the snapshots for missionlist', () => {
        const mission = {
            mission_name: 'Test Mission',
            description: 'Test mission description',
            mission_id: '123',
            member: true,
        };
        const { container } = render(
            <Provider store={store}>
                <MissionList mission={mission} />
            </Provider>,
        );
        expect(container).toMatchSnapshot();
    });
});
