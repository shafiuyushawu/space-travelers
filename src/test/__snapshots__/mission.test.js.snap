// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing mission components Generating the snapshots for missionlist 1`] = `
<div>
  <table>
    <tbody
      class="border"
    >
      <tr>
        <th
          class="border  p-2"
        >
          Test Mission
        </th>
        <td
          class=" border p-2"
        >
          <p>
            Test mission description
          </p>
        </td>
        <td
          class="border p-2 w-36"
        >
          <span
            class="bg-[#00e5ff] font-bold rounded-xl p-1 text-white"
          >
            Active Member
          </span>
        </td>
        <td
          class="border p-2  w-36"
        >
          <button
            class="  border-2 border-[#ff0044] text-[#ff0044] font-bold p-1"
            type="button"
          >
            Leave Mission
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
